src/esw/brushed_motors/Controller.cpp

```diff
@@ -76,15 +94,117 @@ void Controller::moveOpenLoop(float input) {
 
         I2C::transact(deviceAddress, motorIDRegMask | OPEN_PLUS_OP, OPEN_PLUS_WB,
                       OPEN_PLUS_RB, buffer, UINT8_POINTER_T(&angle));
-        currentAngle = (float) ((((float) angle / quadCPR) * 2 * M_PI) - M_PI);
+        currentAngle = (float) (((float) angle / quadCPR) * 2 * M_PI);
     } catch (IOFailure& e) {
         ROS_ERROR("moveOpenLoop failed on %s", name.c_str());
     }
 }
 
+// REQUIRES: valid angle
+// MODIFIES: currentAngle. Also makes controller live if not already.
+// EFFECTS: I2C bus, Sends an close loop command scaled to PWM limits
+// based on allowed voltage of the motor. Also updates angle.
+void Controller::moveClosedLoop(float targetAngle) {
+    try {
+        makeLive();
+
+        auto targetPosition = (int32_t) ((quadCPR * targetAngle) / (2 * M_PI));
+
+        uint8_t buffer[4];
+        memcpy(buffer, UINT8_POINTER_T(&targetPosition), sizeof(targetPosition));
+        int32_t angle;
+
+        I2C::transact(deviceAddress, motorIDRegMask | CLOSED_PLUS_OP, CLOSED_PLUS_WB,
+                      CLOSED_PLUS_RB, buffer, UINT8_POINTER_T(&angle));
+        currentAngle = (float) (((float) angle / quadCPR) * 2 * M_PI);
+    } catch (IOFailure& e) {
+        ROS_ERROR("moveClosedLoop failed on %s", name.c_str());
+    }
+}
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: I2C bus, returns if the MCU is calibrated
+bool Controller::isCalibrated() {
+    uint8_t calibration_status;
+
+    try {
+        makeLive();
+        if (isControllerCalibrated) return true;
+
+        I2C::transact(deviceAddress, motorIDRegMask | IS_CALIBRATED_OP, IS_CALIBRATED_WB,
+                      IS_CALIBRATED_RB, nullptr, UINT8_POINTER_T(&calibration_status));
+
+        isControllerCalibrated = calibration_status;
+
+    } catch (IOFailure& e) {
+        ROS_ERROR("isCalibrated failed on %s", name.c_str());
+        return false;
+    }
+
+    return calibration_status;
+}
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: I2C bus, enables or disables limit switches
+void Controller::enableLimitSwitches(bool enable) {
+    try {
+        makeLive();
+
+        limitAEnable = limitAPresent && enable;
+
+        uint8_t buffer[1];
+        if (!limitAPresent) {
+            memcpy(buffer, UINT8_POINTER_T(&limitAEnable), sizeof(limitAEnable));
+            I2C::transact(deviceAddress, motorIDRegMask | ENABLE_LIMIT_A_OP, ENABLE_LIMIT_A_WB,
+                          ENABLE_LIMIT_A_RB, buffer, nullptr);
+        }
+
+        limitBEnable = limitBPresent && enable;
+
+        if (!limitBPresent) {
```

Comment:
I think this should not have the 'not' in the if statement (same with line 158 above). Sorry if I haven't said this before, I think we can also create a function to handle the updating for one limit switch and then just call that twice, passing the necessary variables.